corehq/sql_db/util.py

```diff
@@ -121,6 +123,20 @@ def get_db_alias_for_partitioned_doc(partition_value):
     return db_name
 
 
+def new_id_in_same_dbalias(partition_value):
+    """
+    Returns a new partition value that belongs to the same db alias as
+        the given partition value does
+    """
+    old_db_name = get_db_alias_for_partitioned_doc(partition_value)
+    new_db_name = None
+    while old_db_name != new_db_name:
+        # todo; guard against infinite recursion
+        new_partition_value = unicode(uuid.uuid4())
+        new_db_name = get_db_alias_for_partitioned_doc(new_partition_value)
+    return new_partition_value
```

Comment:
I did some test on NIC and got a max of 129 iterations and max timing of 430 ms (did 500k tries to get data). The average is 10 iterations so this approach seems fine.